src/index.d.ts

```diff
@@ -279,8 +287,10 @@ export function push(args: {
   ref?: string,
   remote?: string,
   url?: string,
-  authUsername?: string,
-  authPassword?: string
+  username?: string,
+  password?: string,
+  token?: string,
+  oauth2format?: string,
```

Comment:
Should we specifically type `oauth2format` as `'github'|'bitbucket'|'gitlab'`? It will error in compile time, which will prevent from deferring the bug discovery til runtime, but we will still have the `default` check for non-TS use.

Reply:
Yeah, explicit deprecation is the way to go I think. :-)